tests/unit/PeopleAndRoles.spec.ts

```diff
@@ -207,79 +199,274 @@ describe('People And Roles component for Correction', () => {
+
     await wrapper.find(gpAddPerson).trigger('click')
+
+    // verify properties
+    expect(wrapper.vm.$data.activeIndex).toBe(NaN)
     expect(wrapper.vm.$data.isAddingEditingOrgPerson).toBe(true)
+    expect(wrapper.vm.$data.currentOrgPerson.roles).toEqual([{ roleType: 'Partner', appointmentDate: '2022-11-24' }])
+    expect(wrapper.vm.$data.currentOrgPerson.officer.partyType).toBe('person')
+    expect(wrapper.vm.$data.currentOrgPerson.actions).toEqual(['ADDED'])
+
     wrapper.destroy()
   })
 
-  it('sets the data attributes as expected when Add Corporation button is clicked', async () => {
+  it('sets the properties as expected when Add Corporation button is clicked', async () => {
     store.state.stateModel.peopleAndRoles.orgPeople = []
     const wrapper = wrapperFactory()
+
     await wrapper.find(gpAddCorp).trigger('click')
+
+    // verify properties
+    expect(wrapper.vm.$data.activeIndex).toBe(NaN)
     expect(wrapper.vm.$data.isAddingEditingOrgPerson).toBe(true)
+    expect(wrapper.vm.$data.currentOrgPerson.roles).toEqual([{ roleType: 'Partner', appointmentDate: '2022-11-24' }])
+    expect(wrapper.vm.$data.currentOrgPerson.officer.partyType).toBe('organization')
+    expect(wrapper.vm.$data.currentOrgPerson.actions).toEqual(['ADDED'])
+
     wrapper.destroy()
   })
 
   it('shows the add person form when Add Person button is clicked', async () => {
     store.state.stateModel.peopleAndRoles.orgPeople = []
     const wrapper = wrapperFactory()
+
     await wrapper.find(gpAddPerson).trigger('click')
+
     // verify buttons are now disabled
     expect(wrapper.find(gpAddPerson).attributes('disabled')).toBe('disabled')
     expect(wrapper.find(gpAddCorp).attributes('disabled')).toBe('disabled')
+
     // check form
     expect(wrapper.find(orgPersonForm).exists()).toBe(true)
     expect(wrapper.find('.add-person-header').text()).toBe('Add Person')
+
     wrapper.destroy()
   })
 
   it('shows the add corporation form when Add Corporation button is clicked', async () => {
     store.state.stateModel.peopleAndRoles.orgPeople = getOrgList(['partner'])
     const wrapper = wrapperFactory()
     await wrapper.find(gpAddCorp).trigger('click')
+
     // verify buttons are now disabled
     expect(wrapper.find(gpAddPerson).attributes('disabled')).toBe('disabled')
     expect(wrapper.find(gpAddCorp).attributes('disabled')).toBe('disabled')
+
     // check form
     expect(wrapper.find(orgPersonForm).exists()).toBe(true)
     expect(wrapper.find('.add-org-header').text()).toBe('Add Business or Corporation')
+
     wrapper.destroy()
   })
+
+  it('resets state properties correctly', async () => {
+    store.state.stateModel.peopleAndRoles.orgPeople = [
+      {
+        officer: null,
+        roles: [],
+        mailingAddress: null,
+        actions: ['REMOVED', 'REPLACED']
+      }
+    ]
+    const wrapper: any = shallowMount(PeopleAndRoles, {
+      data: () => ({ isAddingEditingOrgPerson: true }),
```

Comment:
This is one way to set a local variable.

Another way is to use `wrapper.setData({ isAddingEditingOrgPerson: true })`.

Reply:
Does `wrapper.setData({ })` actually work with wrapper factory? I was having problems with it in the create UI and had to resort to the method you used.

Reply:
We have lots of implementations of `wrapperFactory()`. Some of them set some state (store) data. Many of them set props data. At a glance, I don't see any that set local variables (aka data).

`wrapper.setData({ })` should work on a wrapper created by a factory function. Sometimes you have to use this anyway if you want to change data after some other changes (ie, not the local variable's initial value).